bliss/encoder.py

```diff
@@ -74,47 +79,38 @@ def forward(self, x):
             ".forward() method for Encoder not available. Use .max_a_post() or .sample()."
         )
 
-    def sample(self, image_ptiles, n_samples):
-        raise NotImplementedError("Sampling from Encoder not yet available.")
-
-    def max_a_post(self, image: Tensor, background: Tensor) -> TileCatalog:
-        """Get maximum a posteriori of catalog from image padded tiles.
-
-        Note that, strictly speaking, this is not the true MAP of the variational
-        distribution of the catalog.
-        Rather, we use sequential estimation; the MAP of the locations is first estimated,
-        then plugged-in to the binary and galaxy encoders. Thus, the binary and galaxy
-        encoders are conditioned on the location MAP. The true MAP would require optimizing
-        over the entire catalog jointly, but this is not tractable.
-
-        Args:
-            image: An astronomical image,
-                with shape `n * n_bands * h * w`.
-            background: Background associated with image,
-                with shape `n * n_bands * h * w`.
-
-        Returns:
-            A dictionary of the maximum a posteriori
-            of the catalog in tiles. Specifically, this dictionary comprises:
-                - The output of LocationEncoder.max_a_post()
-                - 'galaxy_bools', 'star_bools', and 'galaxy_probs' from BinaryEncoder.
-                - 'galaxy_params' from GalaxyEncoder.
-        """
-        assert isinstance(self.map_n_source_weights, Tensor)
+    def sample(
+        self, image: Tensor, background: Tensor, n_samples: Optional[int] = None
```

Comment:
it might be better to have two separate methods in the encoder just like for the location_encoder to avoid the branching, one for MAP and one for samples. But this doesn't have to be in this PR.

Reply:
My thinking was actually to change location encoder to just have one function for both the MAP and sampling to reduce duplication... But I agree let's discuss in a separate PR